faiss_vector_posting.go

```diff
@@ -425,9 +425,11 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, requiresFiltering bool
 						if _, exists := centroidVecIDMap[centroidID]; !exists {
 							centroidVecIDMap[centroidID] = roaring.NewBitmap()
 						}
+						vecIDsUint32 := make([]uint32, 0, len(vecIDs))
 						for _, vecID := range vecIDs {
-							centroidVecIDMap[centroidID].Add(uint32(vecID))
```

Comment:
Consider: since the cluster assignment for an index is fixed, cache it like we do with the vecDocID map, if this PR shows some improvement in perf tests?

Reply:
yes good idea

Reply:
Have you confirmed this to show improvements in perf at the unit level? (considering that the last set of changes that had similar improvements also didn't show any difference at all).